Metrics visualisation (#2)
membraneframework/membrane_rtc_engine_timescaledb: README.md

 * `cleanup_interval` (default: 1 hour) is the number of seconds between database cleanups 
 * `metrics_lifetime` (default: 24 hours) is the number of seconds that must pass from creation before each metric can be deleted during cleanup

## Metrics visualisation with Grafana

Metrics stored in the database using `membrane_rtc_engine_timescaledb` can be simply visualized using Grafana.
To start the dashboard with RTC Engine metrics, mount configuration files, that are in `grafna/provisioning` to the Grafana docker container.

> Eg. if you have cloned this repository in the `/root` directory, add `-v /root/membrane_rtc_engine_timescaledb/grafana/provisioning:/etc/grafana/provisioning` option to your `$ docker run`. Beyond that, you have also set four environment variables for your docker container

Reviewer: I'd avoid using the exact number of variables to set in case that changes in future:

```suggestion
Eg. if you have cloned this repository in the `/root` directory, add `-v /root/membrane_rtc_engine_timescaledb/grafana/provisioning:/etc/grafana/provisioning` option to your `$ docker run`. Beyond that, you have to set the following environment variables for your docker container:
```

 * `DB_NAME` - name of the database, that you store your metrics in
 * `DB_USERNAME` - username used to log into the database
 * `DB_PASSWORD` - password used to log into the database
 * `DB_URL` - database URL in the form of `host:port`, eg. `localhost:5432`

> If you want to take a look at some examples, see `deploy` step in any of the GitHub workflows in [membrane_videoroom](https://github.com/membraneframework/membrane_videoroom/tree/master/.github/workflows) with [docker-compose.yml](https://github.com/membraneframework/membrane_videoroom/blob/master/docker-compose.yml) file used by it.

Reviewer 2: ```suggestion
If you want to take a look at some examples, see `deploy` step in any of the GitHub workflows in [membrane_videoroom](https://github.com/membraneframework/membrane_videoroom/tree/master/.github/workflows) and [docker-compose.yml](https://github.com/membraneframework/membrane_videoroom/blob/master/docker-compose.yml) file used by it.
```

Reviewer 2: I would also consider adding an exemplary permlink to one of these 'deploy steps' in workflows

 * `cleanup_interval` (default: 1 hour) is the number of seconds between database cleanups 
 * `metrics_lifetime` (default: 24 hours) is the number of seconds that must pass from creation before each metric can be deleted during cleanup

## Metrics visualisation with Grafana

Metrics stored in the database using `membrane_rtc_engine_timescaledb` can be simply visualized using Grafana.

> To start the dashboard with RTC Engine metrics, pass volume with the content of `priv/grafana/provisioning` to the Grafana docker container.

Reviewer 2: How about writing a a brief description of what does the `priv/grafana/provisioning` contain?

> You can see, how it is done in this [docker-compose.yml](https://github.com/membraneframework/membrane_videoroom/blob/metrics-visualisation/docker-compose.yml)

Reviewer 2: ```suggestion
You can see, how it is done with the use of the `grafana-provisioning` volume in this [docker-compose.yml](https://github.com/membraneframework/membrane_videoroom/blob/metrics-visualisation/docker-compose.yml)
```

Reviewer 2: The "suggested change" proposed in the comment above is one of the approaches I suggest using so that to help the user understand what is going on. The other approach might be to move the link to the exemplary docker-compose file to the bottom of this section, and emphasize that this configuration sets both the volume and the environmental variables.

> ***NOTE 1*** If you will release a project with dependency to `membrane_rtc_engine_timescaledb`, content of `priv/grafana/provisioning` directory will be contained in the image 

Reviewer 2: ```suggestion
***NOTE 1*** If you release a project with dependency to `membrane_rtc_engine_timescaledb`, content of `priv/grafana/provisioning` directory will be contained in the image 
```

Reviewer 2: BTW - wouldn't it be better to put those **NOTE**'s at the bottom of the section, after the environmental variables specification?

 * `cleanup_interval` (default: 1 hour) is the number of seconds between database cleanups 
 * `metrics_lifetime` (default: 24 hours) is the number of seconds that must pass from creation before each metric can be deleted during cleanup

## Metrics visualisation with Grafana

Metrics stored in the database using `membrane_rtc_engine_timescaledb` can be simply visualized using Grafana.

> To start the dashboard with RTC Engine metrics, create a volume with Grafana configuration files placed in `priv/grafana/provisioning` and pass it to the Grafana container. 

Reviewer 2: ```suggestion
To start the dashboard with RTC Engine metrics, create a volume with Grafana configuration files that are placed in `priv/grafana/provisioning` and mount it in the Grafana container at the default location: `/etc/grafana/provisioning`. 
```

> If you will release an Elixir project with dependency to `membrane_rtc_engine_timescaledb`, `priv/grafana/provisioning` will be contained in the Docker image. 

Reviewer 2: ```suggestion
If you release an Elixir project with dependency to `membrane_rtc_engine_timescaledb`, `priv/grafana/provisioning` will be contained in the Docker image. 
```
I would say that 1st conditional is desired there

## Metrics visualisation with Grafana

Metrics stored in the database using `membrane_rtc_engine_timescaledb` can be simply visualized using Grafana.
To start the dashboard with RTC Engine metrics, create a volume with Grafana configuration files taken from `priv/grafana/provisioning` and mount it in the Grafana container at the default location: `/etc/grafana/provisioning`. 

> If you release an Elixir project with dependency to `membrane_rtc_engine_timescaledb`, `priv/grafana/provisioning` will be contained inside a assembled release, in the `lib/membrane_rtc_engine_timescaledb-0.1.0` directory. 

Reviewer: ```suggestion
When assembling an Elixir release with a dependency on `membrane_rtc_engine_timescaledb` the config files will be put in `lib/membrane_rtc_engine_timescaledb-0.1.0/priv/grafana/provisioning` directory inside the release. 
```

> If you want to have Grafana configuration files in a different path, that does not contain a library version, that can change over time, use `Membrane.RTC.Engine.TimescaleDB.GrafanaHelper.cp_grafana_directory/1`, in one of the release steps in `mix.exs`, as it is done [there](https://github.com/membraneframework/membrane_videoroom/blob/dac1bf06d7130116da038f3b33ff4dc4641a18c6/mix.exs#L15). 

Reviewer: ```suggestion
If you want to put them in a different location (e.g. one that doesn't contain the library version, possibly changing over time), create an additional release step in `mix.exs` and use `Membrane.RTC.Engine.TimescaleDB.GrafanaHelper.cp_grafana_directory/1` inside.
```
I'd remove the link and put the code fragment instead because it can get outdated (like in the case of the link put here).

I don't know how to escape code inside GitHub's suggestion, so here's the snippet:

```elixir
defmodule Example.MixProject do
  use Mix.Project

  def project do
    [
      project: :example_project,
      ...
      releases: [
        example_project: [
          steps: [:assemble, &cp_grafana_config/1]
        ]
      ]
    ]
  end

  defp copy_grafana_config(release) do
    Membrane.RTC.Engine.TimescaleDB.GrafanaHelper.cp_grafana_directory(release.path)
    release
  end
end
```

## Metrics visualisation with Grafana

Metrics stored in the database using `membrane_rtc_engine_timescaledb` can be simply visualized using Grafana.

> To start the dashboard with RTC Engine metrics, create a volume with Grafana configuration files that are placed in `priv/grafana/provisioning` and mount it in the Grafana container at the default location: `/etc/grafana/provisioning`. 

Reviewer: ```suggestion
To start the dashboard with RTC Engine metrics, create a volume with Grafana configuration files taken from `priv/grafana/provisioning` and mount it in the Grafana container at the default location: `/etc/grafana/provisioning`. 
```

> If you release an Elixir project with dependency to `membrane_rtc_engine_timescaledb`, `priv/grafana/provisioning` will be contained in the Docker image. 

Reviewer: 1. Elixir projects are not always released by creating Docker image
2. The important thing here is the info about where that directory will be
3. Move the info about GrafanaHelper here so it is clear what path has to be put in volume's target

> To create a volume containing `priv/grafana/provisioning`, declare it in your `docker-compose.yml` 

Reviewer: This sentence sound like the only way to create that volume is docker-compose. I'd slightly change it to sound more like: here's an example on how to do this with docker-compose.

Metrics stored in the database using `membrane_rtc_engine_timescaledb` can be simply visualized using Grafana.
To start the dashboard with RTC Engine metrics, mount configuration files, that are in `grafna/provisioning` to the Grafana docker container.

> Eg. if you have cloned this repository in the `/root` directory, add `-v /root/membrane_rtc_engine_timescaledb/grafana/provisioning:/etc/grafana/provisioning` option to your `$ docker run`. Beyond that, you have to set the following environment variables for your docker container

Reviewer: 1. This path became obsolete
2. You can also mention the helper you've created
